wlauto/workloads/msword/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,376 @@
+        if (!webview.waitForExists(WAIT_TIMEOUT_5SEC)) {
+            throw new UiObjectNotFoundException("Not found: Sign-in WebView");
+        }
+        emailField = new UiObject(new UiSelector().className(CLASS_EDIT_TEXT).instance(0));
+        int textLength = 30;
+        emailField.clickBottomRight();
+        while (textLength > 0) {
+            getUiDevice().pressDelete();
+        }
+        getUiDevice().waitForIdle();
+        emailField.click();
+        emailField.setText(loginEmail);
+        UiObject passwordField = new UiObject(new UiSelector().className(CLASS_EDIT_TEXT).instance(1));
+        passwordField.setText(loginPass);
+        clickUiObject(BY_DESC, "Sign in", CLASS_BUTTON, true);
+    }
+
+    public void testCreateDocument() throws Exception {
+        newDocument("Newsletter");
+        // Dismiss tooltip if it appears
+        UiObject tooltip = new UiObject(new UiSelector().textContains("Got it").className(CLASS_BUTTON));
+        startLogger("document_tooltip");
+        if (tooltip.waitForExists(WAIT_TIMEOUT_5SEC)) {
+            tooltip.click();
+        }
+        stopLogger("document_tooltip");
+
+        // show command palette
+        UiObject paletteToggle = new UiObject(new UiSelector().resourceId(packageID + "paletteToggleButton"));
+
+        // Rename document
+        startLogger("rename_document");
+        clickUiObject(BY_ID, packageID + "DocTitlePortrait");
+        clickUiObject(BY_ID, packageID + "OfcActionButton1"); // the 'X' button to clear
+        UiObject nameField = getUiObjectByResourceId(packageID + "OfcEditText");
+        nameField.setText(documentName);
+        getUiDevice().pressEnter();
+        stopLogger("rename_document");
+
+        findText("Newsletter Title");
+        startLogger("palette_toggle_show");
+        paletteToggle.click();
+        stopLogger("palette_toggle_show");
+
+        // Text formatting
+        clickUiObject(BY_DESC, "Bold", CLASS_TOGGLE_BUTTON);
+        UiObject undoButton = clickUiObject(BY_DESC, "Undo", CLASS_BUTTON);
+        startLogger("format_bold_italic_underline");
+        clickUiObject(BY_DESC, "Bold", CLASS_TOGGLE_BUTTON);
+        clickUiObject(BY_DESC, "Italic", CLASS_TOGGLE_BUTTON);
+        clickUiObject(BY_DESC, "Underline", CLASS_TOGGLE_BUTTON);
+        stopLogger("format_bold_italic_underline");
+        // Font size
+        UiObject fontSize = new UiObject(new UiSelector().resourceId(packageID + "fsComboBoxButton").instance(1));
+        startLogger("format_font_size_menu");
+        fontSize.click();
+        UiScrollable list = new UiScrollable(new UiSelector().resourceId(
+            packageID + "galleryListControl").childSelector(new UiSelector().className("android.widget.ScrollView")));
+        list.scrollIntoView(new UiSelector().textContains("36"));
+        stopLogger("format_font_size_menu");
+        startLogger("format_font_size_action");
+        // Colours
```

Comment:
Shouldn't this comment be a couple of lines lower? Whilst you are at it, it would make it more readable to have a gap between each text formatting section.